### DISCUSSION TITLE
Nested routes

### DISCUSSION THREAD

**Discussion by neko2h** [nuxt/nuxt#16259] [Questions]
Hi guys. I cant figure out how to make a nested routes.
so i have 3 routes:

- /
- /catalog/
- /catalog/:slug

And it works. But the slug route is not nested. Can you help me please to undestand what i am doing wrong.

https://stackblitz.com/edit/nuxt-starter-wcqaay

**Comment by fathi91** [comment]
Hi,
when you want to have nested routes you should use the built-in component `<NuxtPage />` and you place this component  where you want your nested page to show up.

this is how your page folder should look like:
`pages/index.vue`
`pages/catalog.vue`
`pages/catalog/[slug]/index.vue`

full example: https://stackblitz.com/edit/nuxt-starter-esfemt

**Reply by neko2h** [reply]
> Hi, when you want to have nested routes you should use the built-in component `<NuxtPage />` and you place this component where you want your nested page to show up.
> 
> this is how your page folder should look like: `pages/index.vue` `pages/catalog.vue` `pages/catalog/[slug]/index.vue`
> 
> full example: https://stackblitz.com/edit/nuxt-starter-esfemt

Thank you for your answer. I was probably incorrect in discussion theme. I dont need to show /catalog/:slug  inside /catalog/ . I need to show it as a separate view. I should probably write more precisely. So i have a navigation component with a link /catalog. And i want to set it's class to 'active', when route is /catalog and /catalog/:slug. And i guess i need to extend the default router, but i dont know how.

**Reply by fathi91** [reply]
you should take a look at https://router.vuejs.org/guide/migration/#removal-of-the-exact-prop-in-router-link and https://github.com/vuejs/rfcs/blob/master/active-rfcs/0028-router-active-link.md#summary.
